src/blocks/weather/open_weather_map.rs

```diff
@@ -79,24 +95,184 @@ struct ApiWeather {
+                Some(place) => {
+                    let url = format!("{GEO_URL}/direct?q={place}&appid={api_key}");
+                    REQWEST_CLIENT
+                        .get(url)
+                        .send()
+                        .await
+                        .error("Geo request failed")?
+                        .json::<Vec<CityCoord>>()
+                        .await
+                        .error("Geo failed to parse json")?
+                        .first()
+                        .and_then(|city| {
+                            let location = Some(format!("lat={}&lon={}", city.lat, city.lon));
+                            *LOCATION_CACHE.lock().unwrap() = location.clone();
+                            location
+                        })
+                }
+                None => None,
+            },
+        };
+
+        location_query = match location_query {
+            Some(x) => Some(x),
+            None => match self.config.zip.as_ref() {
+                Some(zip) => {
+                    let url = format!("{GEO_URL}/zip?zip={zip}&appid={api_key}");
+                    let city: CityCoord = REQWEST_CLIENT
+                        .get(url)
+                        .send()
+                        .await
+                        .error("Geo request failed")?
+                        .json()
+                        .await
+                        .error("Geo failed to parse json")?;
+
+                    let location = Some(format!("lat={}&lon={}", city.lat, city.lon));
+                    *LOCATION_CACHE.lock().unwrap() = location.clone();
+                    location
+                }
+                None => None,
+            },
+        };
+
+        location_query = match location_query {
+            Some(x) => Some(x),
+            None => match self.config.city_id.as_ref() {
+                Some(city_id) => {
+                    let city_id = &city_id.parse().error("city_id must be numeric")?;
+
+                    let mut cities_temp_path = env::temp_dir();
+                    let user = env::var("USER");
+                    cities_temp_path.push(format!(
+                        "current.city.list.min.json.gz-{}",
+                        user.as_deref().unwrap_or("no-user")
+                    ));
+
+                    let cities_temp_timestamp = fs::metadata(&cities_temp_path)
+                        .await
+                        .and_then(|m| m.modified())
+                        .map(|sys_time| {
+                            DateTime::<Utc>::from(sys_time)
+                                .format("%a, %d %b %Y %H:%M:%S GMT")
+                                .to_string()
+                        });
+
+                    let cities_request = if let Ok(timestamp) = cities_temp_timestamp.as_ref() {
+                        debug!("{}", timestamp);
+                        REQWEST_CLIENT.get(CITIES_URL).header(
+                            reqwest::header::IF_MODIFIED_SINCE,
```

Comment:
This isn't being respected by owm's server. Curl is smart enough to handle it, but reqwest/hyper doesn't yet.